src/App.tsx

```diff
@@ -1,24 +1,78 @@
 import React from 'react';
 import './App.scss';
+import Clock from './Clock';
 
-const App: React.FC = () => {
-  setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean;
+  clockName: number;
 };
 
+class App extends React.Component<{}, State> {
+  state: State = {
+    isClockVisible: true,
+    clockName: 0,
+  };
+
+  setRandomName = () => {
+    if (this.state.isClockVisible) {
+      this.setState((prevState) => {
+        const newName = Math.trunc(Math.random() * 10 + 1);
+
+        // eslint-disable-next-line no-console
+        console.log(`The Clock was renamed from ${prevState.clockName} to ${newName}`);
+
+        return {
+          clockName: newName,
+        };
+      });
+    }
+  };
+
+  showClock = () => {
+    this.setState({ isClockVisible: true });
+  };
+
+  hideClock = () => {
+    this.setState({ isClockVisible: false });
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+    const { setRandomName, hideClock, showClock } = this;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {isClockVisible
+          && (<Clock name={clockName} />)}
```

Comment:
```suggestion
        {isClockVisible && (
          <Clock name={clockName} />
        )}
```
Pay attention to the formatting